File: 1208/1208.py
# -*- coding:utf-8 -*-
# Author: washing
# DateTime: 2021/2/5 12:07
# File: 1208.py
# Desc: 
class Solution:
    def equalSubstring(self, s: str, t: str, maxCost: int) -> int:
        # totalCost = 0
        # li = []
        # count = 0
        # for c1, c2 in zip(s, t):
        #     num = abs(ord(c1) - ord(c2))
        #     totalCost += num
        #     li.insert(0, num)
        #     while totalCost > maxCost:
        #         totalCost -= li.pop()
        #     count = max(count, len(li))
        # return count
        l = r = 0
        totalCost = 0
        for c1, c2 in zip(s, t):
            r += 1
            num = abs(ord(c1) - ord(c2))
            totalCost += num
            if totalCost > maxCost:
                totalCost -= abs(ord(s[l])-ord(t[l]))
                l += 1
        return r - l

File: Solutions/1208/1208.java
/**
 * @Author: washing
 * @DateTime: 2021/02/5 12:06
 * @File: 1208.java
 * @Desc: 
 **/
class Solution {
    public int equalSubstring(String s, String t, int maxCost) {
        int l = 0;
        int r = 0;
        int totalCost = 0;
        while (r < s.length()){
            totalCost += Math.abs(s.charAt(r)-t.charAt(r));
            if (totalCost > maxCost){
                totalCost -= Math.abs(s.charAt(l)-t.charAt(l));
                l++;
            }
            r++;
        }
        return r-l;
    }
}